Intersections_3/include/CGAL/Intersections_3/internal/tetrahedron_lines_intersections_3.h

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2019 GeometryFactory(France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org); you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public License as
+// published by the Free Software Foundation; either version 3 of the License,
+// or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0+
+//
+//
+// Author(s)     : Maxime Gimeno
+//
+
+#ifndef TETRAHEDRON_LINES_INTERSECTIONS_3_H
+#define TETRAHEDRON_LINES_INTERSECTIONS_3_H
+
+#include <vector>
+#include <CGAL/kernel_basic.h>
+#include <CGAL/intersections.h>
+
+namespace CGAL {
+
+namespace Intersections {
+
+namespace internal {
+template<class K, class O>
+struct Tetrahedron_lines_intersection_3_base
+{
+  typedef typename Intersection_traits<K,
+      CGAL::Tetrahedron_3<K>,
+      O>::result_type Result_type;
+
+  typedef typename Intersection_traits<K,
+      CGAL::Triangle_3<K>,
+      O>::result_type Inter_type;
+
+  Tetrahedron_lines_intersection_3_base(const typename K::Tetrahedron_3& tet,
+  const O& o):tet(tet), o(o)
+  {}
+  const typename K::Tetrahedron_3& tet;
+  const O& o;
+  std::vector<typename K::Point_3> res_points;
+  Result_type output;
+  virtual bool all_inside_test()
+  {
+    return false;
+  }
+
+  virtual bool are_extremities_inside_test()
+  {
+    return false;
+  }
+  virtual ~Tetrahedron_lines_intersection_3_base()
```

Comment:
Why is this virtual?

Reply:
A class that has at least a virtual function must declare a virtual destructor.